stumpy/core.py

```diff
@@ -659,16 +659,43 @@ def welford_nanstd(a, w=None):
     return np.sqrt(np.clip(welford_nanvar(a, w), a_min=0, a_max=None))
 
 
-def rolling_nanstd(a, w):
+@njit(parallel=True, fastmath={"nsz", "arcp", "contract", "afn", "reassoc"})
+def _rolling_nanstd_1d(a, w):
     """
-    Compute the rolling standard deviation for 1-D and 2-D arrays while ignoring NaNs
-    using a modified version of Welford's algorithm but is much faster than using
-    `np.nanstd` with stride tricks.
+    A Numba JIT-compiled and parallelized function for computing the std of
+    each subsequence with length `w` in `a`, which is a 1D array.
 
-    This a convenience wrapper around `welford_nanstd`.
+    Parameters
+    ----------
+    a : numpy.ndarray
+        The input 1D array
+
+    w : int
+        The rolling window size
+
+    Returns
+    -------
+    out : numpy.ndarray
+        This 1D array has the length of `a.shape[0]-w+1`. `out[i]`
+        contains the std value of `a[i : i + w]`
+    """
+    l = a.shape[0] - w + 1
+    out = np.empty(l, dtype=np.float64)
```

Comment:
I think it's fine to combine the two lines into `out = np.empty(a.shape[0] - w + 1, dtype=np.float64)` as it is pretty straightforward

Reply:
Just to make sure we are on the same page, I assume you already noticed `l` in the next line in `prange(l)`, and you are okay with replacing that with `a.shape[0] - w + 1.`

Is that correct?

Reply:
Ahh, I didn't notice that. In that case, please ignore this comment